hail/python/hail/methods/qc.py

```diff
@@ -491,13 +504,128 @@ def n_discordant(counter):
     return glob, per_sample.cols(), per_variant.rows()
 
 
+def _service_vep(ht, config, block_size, csq, tolerate_parse_error):
+    token = secret_alnum_string(16)
+
+    backend = hl.context.current_backend()
+    assert isinstance(backend, hl.backend.service_backend.ServiceBackend)
+
+    tmp_dir = hl.backend.service_backend.tmp_dir()
+
+    vep_input_path = f'{tmp_dir}/vep/inputs/{token}.vcf'
+    vep_output_path = f'{tmp_dir}/vep/outputs/{token}'
+
+    hl.export_vcf(ht, vep_input_path, parallel='header_per_shard')
```

Comment:
Use `hl.TemporaryFile` or `hl.TemporaryDirectory`. They are context managers which will take care of deleting the temporary file or directory.